### PR TITLE
[diffusion] model: update to new model format

### PR DESCRIPTION
**Pull request by nv-dmajchrowski**
## Motivation

PR [!24994](https://github.com/sgl-project/sglang/pull/24994) added initial support for for a new world model (not yet released).  The final format slightly changed, this PR updates the codebase to work with the updated checkpoint.
Additionally, we removed custom guardrails codebase and switched to package driven approach (must be installed manually via pip install before running the model).

## Modifications

- Support for new diffusers format (the one that will be released)
- Safety checker managed via guardrail package
- Note about missing support for sound/action/v2v modalities

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26610929782](https://github.com/sgl-project/sglang/actions/runs/26610929782)
Latest PR Test (Extra): :x: [Run #26610929761](https://github.com/sgl-project/sglang/actions/runs/26610929761)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py [hunk 1]
```diff
@@ -34,413 +22,58 @@
...
-    frames = video[0] if video.ndim == 5 else video
-    frames = _video_guardrail(frames)
-    if video.ndim == 5:
-        frames = frames[np.newaxis]
-    return frames
+    batched = video.ndim == 5
+    frames = video[0] if batched else video
+    result = _checker.check_video_safety(frames)
+    out = result if result is not None else frames
+    return out[np.newaxis] if batched else out
```
[the remark this answers is not in the document]
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py]: Fixed

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py [hunk 2]
```diff
@@ -34,413 +22,58 @@
...
-_text_guardrail: TextGuardrailFn | None = None
-_video_guardrail: VideoGuardrailFn | None = None
-_initialized = False
+    logger.info("Initializing Cosmos3 guardrails (offload_to_cpu=%s) ...", offload_to_cpu)
+    _checker = CosmosSafetyChecker()
+    idle_device = "cpu" if offload_to_cpu else "cuda"
+    for runner in (_checker.text_guardrail, _checker.video_guardrail):
+        for m in runner.models:
+            if isinstance(m, torch.nn.Module):
+                m.to(idle_device)
```
[the remark this answers is not in the document]
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py:49]: Fixed

**Comment by nvpohanh** [comment]
@mickqian Could you help to review this since you have also reviewed https://github.com/sgl-project/sglang/pull/24994 ? Thanks!

**Review by mickqian — APPROVED** [review]

**Comment by nvpohanh** [comment]
@nv-dmajchrowski Please fix the linter issue:

```
From https://github.com/sgl-project/sglang
 * branch                main       -> FETCH_HEAD
Changes under the legacy docs/ directory are not allowed.

The documentation has been migrated. Please make documentation updates in the
corresponding location under docs_new/ instead.

Detected legacy docs/ changes:
  - docs/diffusion/compatibility_matrix.md
```
